spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarPropertiesMapper.java

```diff
@@ -214,6 +227,17 @@ void customizeReaderContainerProperties(PulsarReaderContainerProperties readerCo
 		PulsarProperties.Reader properties = this.properties.getReader();
 		PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
 		map.from(properties::getTopics).to(readerContainerProperties::setTopics);
+		customizeReaderStartupProperties(readerContainerProperties);
+	}
+
+	private void customizeReaderStartupProperties(PulsarReaderContainerProperties readerContainerProperties) {
```

Comment:
It's unfortunate that this is the exact same code as `customizeListenerStartupProperties`? Since the feature is shared, isn't there a way to avoid the code duplication?

Reply:
The code to de-duplicate was more complicated than having this in multiple places.